uniffi_bindgen/src/interface/types/mod.rs

```diff
@@ -156,6 +160,7 @@ impl From<&Type> for FFIType {
             | Type::Duration
             | Type::External { .. } => FFIType::RustBuffer,
             Type::Wrapped { prim, .. } => FFIType::from(prim.as_ref()),
+            Type::DelegateObject(_) => unreachable!("Delegate objects should never cross the FFI"),
```

Comment:
I was surprised that this gained a variant in the `Type` enum. I guess that's to support the synthesized delegate argument in constructors?

Reply:
Gaining a `Type` variant surprised me too! Once I started on error checking, it became a necessity.

e.g. this would be a bit harder to detect as an error:

```
[Decorator]
interface Object1 {}

interface Object1 {}

namespace {
  void set_not_a_decorator(Object1 obj);
}
```

Reply:
We also need to refer to it by type name in the foreign language code (not just lifting and lowering).